### DISCUSSION TITLE
PID Tuning and a maibe stupid G28 question :-)

### DISCUSSION THREAD

**Discussion by Team-Deepflights** [grblHAL/core#144] [General]
Hi,

I finally finished the lathe and it works as it should.

Only threading still makes me mad, the angles are not as they should be and thats why I reflashed the Teensy with PID logging (1000) enabled.
But if I click on the button, and than start a few thread cycles in the air, I see nothing at all on the PID logging.
The lathe have no VFD, I only can change RPM by turning a potentiometer. The RPM are measured with a sensor and the RPM are shown correctly in IOSender.
Could please someone explain me how the logs are visualized? Is there anything more to set up in the program?
How is the workflow to get the data?

The other issue is, I tried to write a macro with homing and set G92 to get a fixed zero to measute the tool offsets.
I used G28, did not know that $H is the way to go.
I am not sure, but I think since then if I look at the offsets I see x-1z-1 at the G28 row.
The tools had an wrong offset too. I corrected them but it seems to "forget" it, just right now I had this issue again, I wanted 12mm diameter and got 13mm.
How can I disable the G28?

**Comment by terjeio** [comment]
> But if I click on the button, and than start a few thread cycles in the air, I see nothing at all on the PID logging.

You have to run a cycle then go to the log tab and click on the button. Only the last cycle is stored in the log.

> I am not sure, but I think since then if I look at the offsets I see x-1z-1 at the G28 row.

G28 is not an offset, it is a an absolute position. I added G28 and G30 to the offset tab in order to make them accessible - a bit misleading I know.

> How can I disable the G28?

It cannot be disabled, only set to 0. I may change that in a later version. G28 (and G30) should not be used unless the machine has been homed.

**Comment by Team-Deepflights** [comment]
After homing I send the command G28 Z0 X0, the lathe run to the coordinates and went back to the endswitches, after this G28 was zero.
I forgot the absolute position thing, thank you for open my eyes.

About logging, I did it that way you described, without doing a cycle, I get a dotted line in the middle ( sure, not data), running a threading cycle and return to the PID tuner and click "get data" the window is greyed out ( no feedback)
I compiled with `#define PID_LOG 1000`

**Reply by terjeio** [reply]
Here is an example of what I get:

![image](https://user-images.githubusercontent.com/20260062/171789095-0a4c9651-6012-4f77-abbe-0863221b2802.png)

In the console I see a PID report after pressing "Get data":
`
[PID:100.000,0.000,2|0.050,0.050,0.110,0.110,0.160,0.160,0.212,0.188...`

My test files:
```
m3s200
g0x2
g0z2
g1x0f50
g33z-15k1.5
g0x2
g0z2
M30
```
and
```
G18
M3S200
G8
G0X25
G0Z2
G76 P1.5 Z-10 K1 I-0.1 J0.4 R2 H2 Q29.5
M30
```

My settings:

![image](https://user-images.githubusercontent.com/20260062/171789579-5eaee4de-5472-4d2c-8b61-18d12f7f9545.png)

**Comment by Team-Deepflights** [comment]
I tried your gcodes, the console show a PID, but the PID Tuner window keeps blank.
Graphics are generated, I am able to see the gcode moves.
I have no clue...

**Reply by terjeio** [reply]
Post a screenshot of what you get in the _PID tuner_ tab.
